File: tests/benchmarks/resolution/fixtures/csharp/Validators.cs
using System;

namespace Benchmark;

public static class Validators
{
    public static bool IsValidEmail(string email)
    {
        return !string.IsNullOrEmpty(email) && email.Contains("@");
    }

    public static bool IsValidName(string name)
    {
        return !string.IsNullOrEmpty(name) && name.Length >= 2;
    }

    public static bool ValidateUser(User user)
    {
        return IsValidEmail(user.Email) && IsValidName(user.Name);
    }
}

File: tests/benchmarks/resolution/fixtures/php/Validators.php
<?php

class Validators
{
    public static function isValidEmail(string $email): bool
    {
        return str_contains($email, '@');
    }

    public static function isValidName(string $name): bool
    {
        return strlen($name) >= 2;
    }

    public static function validateUser(User $user): bool
    {
        return self::isValidEmail($user->email) && self::isValidName($user->name);
    }
}
